defender/downloaded/Compound_Governance_Automation.js

```diff
@@ -0,0 +1,143 @@
+const ethers = require('ethers');
+
+const { KeyValueStoreClient } = require('defender-kvstore-client');
+const { DefenderRelayProvider, DefenderRelaySigner } = require('defender-relay-client/lib/ethers');
+
+const compoundGovernanceAddress = '0xc0Da02939E1441F497fd74F78cE7Decb17B66529';
+
+const compoundGovernanceAbi = [
+  'function execute(uint proposalId)',
+  'function initialProposalId() view returns (uint256)',
+  'function proposalCount() view returns (uint256)',
+  'function proposals(uint256) view returns (uint256 id, address proposer, uint256 eta, uint256 startBlock, uint256 endBlock, uint256 forVotes, uint256 againstVotes, uint256 abstainVotes, bool canceled, bool executed)',
+  'function queue(uint proposalId)',
+  'function state(uint256 proposalId) view returns (uint8)',
+];
+
+async function readyToExecute(contract, proposalId) {
+  const block = await contract.provider.getBlock('latest');
+  const proposal = await contract.proposals(ethers.BigNumber.from(proposalId));
+  return (block.timestamp > (proposal.eta).toNumber());
+}
+
+exports.handler = async function handler(autotaskEvent) {
+  // ensure that the autotaskEvent Object exists
+  if (autotaskEvent === undefined) {
+    throw new Error('autotaskEvent undefined');
+  }
+
+  // create a Provider and Signer from the connected Relay
+  console.debug('Creating DefenderRelayProvider');
+  const provider = new DefenderRelayProvider(autotaskEvent);
+  console.debug('Creating DefenderRelaySigner');
+  const signer = new DefenderRelaySigner(autotaskEvent, provider, { speed: 'fast' });
+
+  // Test relay
+  try {
+    await signer.getAddress();
+  } catch (error) {
+    console.error('Relay is not working, check if it is connected : ', error);
+    throw error;
+  }
+
+  // create an ethers.js Contract Object to interact with the on-chain smart contract
+  console.debug('Creating governanceContract');
+  const governanceContract = new ethers.Contract(
+    compoundGovernanceAddress,
+    compoundGovernanceAbi,
+    signer,
```

Comment:
If you want to make this read-only for now, the signer can be replaced with provider